scripts/stabilization.py

```diff
@@ -0,0 +1,207 @@
+    const std::shared_ptr<const FiniteElement> element = J->function_space()->element();
+
+    const Cell cell(*mesh, c.index);
+    const double h = cell.h();
+
+    // get cell vertex coordinates (not coordinate dofs)
+    std::vector<double> vertex_coords;
+    cell.get_vertex_coordinates(vertex_coords);
+    
+    // get cell coordinate dofs not vertex coordinates
+    std::vector<double> coordinate_dofs;
+    cell.get_coordinate_dofs(coordinate_dofs);  
+
+    // tabulate the coordinates of all dofs on an element
+    boost::multi_array<double, 2> coords;
+    element->tabulate_dof_coordinates(coords, vertex_coords, cell);
+
+	
+    dolfin_assert(J->value_size() == J_->value_size());
+    dolfin_assert(J_->value_size() == J_n->value_size());
+	
+	//
+    const unsigned int num_nodes = coords.num_elements();
+    const unsigned int n = J->value_size();
+	
+	
+    std::vector<double> basis_derivatives(dim * num_nodes * num_nodes);
+    std::vector<double> derivatives(num_nodes);
+    
+    double *basis_derivatives_pointer;
+    
+    // buffer arrays for function values
+    Array<double> vals_J(n);
+    Array<double> vals_J_(n);
+    Array<double> vals_J_n(n);
+    
+    Array<double> vals_Enorm(1);
+    Array<double> val_dt(1);
+    //Array<double> val_c_e(1);
+    //Array<double> val_nu_max(1);
+    
+    // aggregated function values
+    Array<double> vals_J_all(n * num_nodes);
+    Array<double> vals_J__all(n * num_nodes);
+    Array<double> vals_J_n_all(n * num_nodes);
+
+	// evaluate normalized entropy variation
+	E_norm->eval(vals_Enorm, x, c);
+	dt->eval(val_dt, x, c);
+	//c_e->eval(val_c_e, x, c);
+	//nu_max->eval(val_nu_max, x, c);
+
+
+    double Dh = 0;
+    double max_Dh = 0;
+    
+   	// 
+    std::vector<double> x_(dim);
+    Array<double> x_function(dim);
+
+	// pointers to (vertex) coordinates for evaluation of basis derivatives
+    double* x_basis = &x_[0];
+    const double *vertex_coords_basis = &vertex_coords[0];
+    
+    // index
+    int j = 0;
+
+    // Different for higher dimension, since all coordinates need to be updated
+    for(auto i = coords.origin(); i < (coords.origin() + coords.num_elements()); ++i)  
+    {
+    	for (unsigned int d=0; d<dim; ++d) {
+	    	x_[d] = i[d];
+    	    x_function[d] = i[d];
+        }
+        
+        // evaluate functions at current dof coordinate
+        J->eval(vals_J, x_function, c);
+        J_->eval(vals_J_, x_function, c);
+        J_n->eval(vals_J_n, x_function, c);
+        
+        // aggregate values
+        for (unsigned int e=0; e<dim; ++e) {
+			vals_J_all[j * dim + e] = vals_J[e];
+			vals_J__all[j * dim + e] = vals_J_[e];
+			vals_J_n_all[j * dim + e] = vals_J_n[e];        	
+        }
+        
+        // pointer to current basis derivatives and 
+        basis_derivatives_pointer = &basis_derivatives[num_nodes * j];
+		element->evaluate_basis_derivatives_all(1, basis_derivatives_pointer, x_basis, vertex_coords_basis, (int) cell.orientation());
```

Comment:
Instead of assigning a pointer you could just pass it when calling the class method:
```c++
element->evaluate_basis_derivatives_all(1,
                                        &basis_derivatives[num_nodes * j],
                                        x_basis,
                                        vertex_coords_basis,
                                        (int)cell.orientation());
```